src/Normalizer/ContentEntityNormalizer.php

```diff
@@ -269,26 +271,16 @@ public function denormalize($data, $class, $format = NULL, array $context = [])
 
     // @todo {@link https://www.drupal.org/node/2599926 Use the passed $class to instantiate the entity.}
 
+    $entity = NULL;
     if ($entity_id) {
-      if ($entity = $storage->load($entity_id) ?: $storage->loadDeleted($entity_id)) {
-        if (!empty($translations[$entity->language()->getId()])) {
-          $translation = $translations[$entity->language()->getId()];
-          unset($translation['default_langcode']);
-          if ($entity_type->hasKey('bundle')) {
-            $this->extractBundleData($translation, $entity_type);
-          }
-          $this->denormalizeFieldData($translation, $entity, $format, $context);
-        }
-      }
-      elseif (isset($translations[$default_langcode][$id_key])) {
+      if (isset($translations[$default_langcode][$id_key])) {
         unset($translations[$default_langcode][$id_key], $translations[$default_langcode][$revision_key]);
         $entity_id = NULL;
 
         $entity = $this->createEntityInstance($translations[$default_langcode], $entity_type, $format, $context);
```

Comment:
As mentioned previously, as long as createEntityInstance is used we're good in the 8.4.x serialization world, as field level denormalization will happen correctly.